app/src/main/java/lu/geoportail/map/LuxTileServer.kt

```diff
@@ -35,71 +47,241 @@ class LuxTileServer(
+            fileOutputStream.write(buf, 0, len)
+        }
+        sourceStream.close()
+        fileOutputStream.close()
+    }
+
+    private fun deleteAssets(mapName: String, basePath: File) {
+        val sourcePaths = getPaths(mapName)
+        for (i in 0 until sourcePaths.length()) {
+            val url = URL(sourcePaths.getString(i))
+            val filename =url.file
+            val file = File(basePath, filename)
+            file.delete()
+        }
+    }
+
+    private fun moveAssets(sourcePaths: JSONArray, fromPath: File, toPath: File) {
+        for (i in 0 until sourcePaths.length()) {
+            val url = URL(sourcePaths.getString(i))
+            val filename =url.file
+            val toFile = File(toPath, filename)
+            if(!toFile.parentFile.exists()) Files.createDirectories(toFile.parentFile.toPath())
+            File(fromPath, filename).renameTo(toFile)
+        }
+    }
+
+    private fun getPaths(ressourceName: String): JSONArray {
+        // metadata will be read from local file until it is available online
+        return try {
+            val url = URL("file:${this.filePath}/dl/versions/$ressourceName.meta")
+            val json = JSONObject(url.openStream().use { it.readBytes() }.toString(Charsets.UTF_8))
+            val paths = json.getJSONArray("sources") ?: JSONArray()
+            paths.put("file:/versions/$ressourceName.meta")
+        } catch (e: FileNotFoundException) {
+            JSONArray()
+        }
+    }
+
+    private fun getVer(ressourceName: String): String? {
+        return try {
+            val url = URL("file:${this.filePath}/dl/versions/$ressourceName.meta")
+            val json = JSONObject(url.openStream().use { it.readBytes()}.toString(Charsets.UTF_8))
+            json.getString("version")
+        }
+        catch (e: FileNotFoundException) {
+            null
+        }
+    }
+
+    private fun getSize(ressourceName: String): String? {
+        return try {
+            val dlPath = when(getStatus(ressourceName)) {
+                "in progress" -> "tmp"
+                else -> "dl"
+            }
+            val url = URL("file:${this.filePath}/$dlPath/versions/$ressourceName.meta")
+            val json = JSONObject(url.openStream().use { it.readBytes()}.toString(Charsets.UTF_8))
+            val src = json.getJSONArray("sources")
+            var totalSize: Long = 0
+            for (i in 0 until src.length()) {
+                try {
+                    val filename = URL(src.getString(i)).file
+                    totalSize += Files.size(
+                        File(
+                            File(this.filePath, dlPath),
+                            filename
+                        ).toPath()
+                    )
+                }
+                catch (e: NoSuchFileException) {}
+            }
+            totalSize.toString()
+        }
+        catch (e: FileNotFoundException) {
+            null
```

Comment:
You should throw a "ResourceNotFound" or something in case the resource in unavailable.

Reply:
yes, I should reorganize the code somehow to catch the exceptions later.
There quite a lot of potential exceptions to handle though, so it's easy to miss a thing. And on some exceptions, the script should be able to continue.